js/run_examples.js

```diff
@@ -0,0 +1,159 @@
+
+        var prepareForMain = function()
+        {
+            var src = $.browser.msie && $.browser.version < 9.0 ? orgSrc[0].innerText : orgSrc.text();
+            var arr = src.split("\n");
+            var str = "";
+            for ( i = 0; i < arr.length; i++)
+            {
+                str += arr[i]+"\n";
+            }
+            if ($.browser.msie && $.browser.version < 9.0)
+                str = str.substr(0, str.length - 1);
+            else
+                str = str.substr(0, str.length - 2);
+
+            return str;
+        };
+
+        var editor = CodeMirror.fromTextArea(thisObj[0], {
+            lineNumbers: true,
+            tabSize: 4,
+            indentUnit: 4,
+            indentWithTabs: true,
+            mode: "text/x-d",
+            lineWrapping: true,
+            theme: "eclipse",
+            readOnly: false,
+            matchBrackets: true
+        });
+
+        editor.setValue(prepareForMain());
+
+        var height = function(diff) {
+            var par = code != null ? code : parent.parent().children("div.d_code");
+            return (parseInt(par.css('height')) - diff) + 'px';
+        };
+        var btnParent = parent.parent().children(".d_example_buttons");
+
+        var runBtn = btnParent.children("input.runButton");
+        var editBtn = btnParent.children("input.editButton");
+        var resetBtn = btnParent.children("input.resetButton");
+
+        var outputDiv = parent.children("div.d_code_output");
+
+        var code = $(editor.getWrapperElement());
+        code.css('display', 'none');
+
+        var output = outputDiv.children("textarea.d_code_output");
+        var outputTitle = outputDiv.children("span.d_code_title");
+
+        var hideAllWindows = function()
+        {
+            outputDiv.css('display', 'none');
+            parent.parent().children("div.d_code").css('display', 'none');
+            code.css('display', 'none');
+        };
+
+        editBtn.click(function(){
+            resetBtn.css('display', 'inline-block');
+            hideAllWindows();
+            code.css('display', 'block');
+            editor.refresh();
+            editor.focus();
+        });
+        resetBtn.click(function(){
+            resetBtn.css('display', 'none');
+            editor.setValue(prepareForMain());
+            hideAllWindows();
+            parent.parent().children("div.d_code").css('display', 'block');
+        });
+        runBtn.click(function(){
+            resetBtn.css('display', 'inline-block');
+            $(this).attr("disabled", true);
+            hideAllWindows();
+            output.css('height', height(31));
+            outputDiv.css('display', 'block');
+            outputTitle.text("Application output");
+            output.html("Running...");
+            output.focus();
+
+            var currentPackage = $('body')[0].id;
+            var codeOut = "void main(){ ";
+            codeOut += "import " + currentPackage + "; ";
```

Comment:
Clever! 

I had a spin with this some time ago.

From what I recall some modules, like std.base64 depend on 3rd party modules like std.stdio and of course do not include the imports in their unittests/examples. This is something we should also consider hm